### ISSUE TITLE
Cannot use latest package:test due to build_test regression

### ISSUE THREAD

**Issue by matanlurey** [dart-lang/angular#1225]
Latest run: https://travis-ci.org/dart-lang/angular/jobs/365356819

Example failure:
```
line 1, column 74 of asset:test_lib/lib/test_lib.dart: Could not resolve annotation for class Example
  @Component(selector: 'example')
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Talking to @natebosch, it's likely the root cause is https://github.com/dart-lang/test/pull/801, which removed the `.packages` file. We don't strictly need `pub run build_runner test -- -p vm` for this repository, so I could either:

* Hard-code fixes for the two failing packages.
* Pin the version of `package:test` for the failing packages.
* Try to fix-forward with the test team.

**Comment by matanlurey** [comment]
@natebosch (offline):

> I'd probably go for pinning pkg/test first
> my suspicion is that after the change `PackageResolver.current` isn't working anymore.

**Comment by matanlurey** [comment]
Partially fixed by #1226.

https://github.com/dart-lang/build/issues/1285 will be the "real" fix.
